vllm/compilation/fusion.py

```diff
@@ -360,11 +470,89 @@ def __call__(self, graph: fx.Graph):
         logger.debug("Replaced %s patterns", self.matched_count)
 
     def uuid(self) -> Any:
-        return self.hash_source(
-            self,
+        fusion_patterns = [
             RMSNormQuantPattern,
             RMSNormStaticQuantPattern,
             RMSNormDynamicQuantPattern,
             FusedAddRMSNormStaticQuantPattern,
             FusedAddRMSNormDynamicQuantPattern,
+        ]
+        if is_rocm_aiter_rmsnorm_enabled():
+            fusion_patterns.extend(
+                [AiterRMSGroupQuantFP8Pattern, AiterFusedAddRMSGroupQuantPattern]
+            )
+        return self.hash_source(self, *fusion_patterns)
+
+
+if current_platform.is_rocm() and envs.VLLM_ROCM_USE_AITER:
+    from .matcher_utils import MatcherAiterFusedMulAdd
+
+    class AiterMulAddFusionPattern:
+        def __init__(self):
+            config = get_current_vllm_config()
+            self.model_dtype = (
+                config.model_config.dtype if config.model_config else None
+            )
+            self.enabled = current_platform.is_rocm() and envs.VLLM_ROCM_USE_AITER
+
+            self.fused_mul_add_matcher = MatcherAiterFusedMulAdd()
+
+        def register(self, pm_pass: PatternMatcherPass):
+            if not self.enabled:
+                return
+
+            def pattern(x: torch.Tensor, a: torch.Tensor, b: torch.Tensor):
+                mul_add = self.fused_mul_add_matcher.forward_native(x, a, b)
+                return mul_add
+
+            def replacement(x: torch.Tensor, a: torch.Tensor, b: torch.Tensor):
+                # AITER's fused_mul_add requires that a either be a
+                # scalar or have the same number of elements as x.
+                if (
+                    isinstance(a, torch.Tensor)
+                    and isinstance(b, torch.Tensor)
+                    and (a.numel() in [1, x.numel()])
+                    and (b.numel() in [1, x.numel()])
+                ):
+                    return self.fused_mul_add_matcher.forward_custom(x, a, b)
+                else:
+                    return self.fused_mul_add_matcher.forward_native(x, a, b)
```

Comment:
Why is this conditional inside the replacement? The tracing will make this always pick a single branch depending on the graph inputs. I think it would be clearer to add an `extra_check` parameter to the `register_replacement` function